Feature/nevisaccessapp 6287 multi user transaction confirmation fixes

balazs-gerlei opened the pull request.

balazs-gerlei committed:
NEVISACCESSAPP-6287: Pass only the filtered (valid) accounts to transaction confirmation and account selector when there are multiple registered accounts

balazs-gerlei committed:
NEVISACCESSAPP-6287: Change the order Account Selection and Transaction Confirmation screens are shown: the former now displayed before the latter

- Transaction Confirmation screen now only deals with a single Account as that should be selected before that
- Transaction Confirmation message is passed along Account Selection now
- Simplified/clarified logic in AccountSelectorImpl
- Add TransactionConfirmationResponse and TransactionConfirmationUseCase and its implementation, the former is used by AccountSelectorImpl and the latter is used by SelectAccountViewModel to show transaction confirmation when a message is received

laszlo-domonkos commented on app/src/main/java/ch/nevis/exampleapp/coroutines/domain/model/response/TransactionConfirmationResponse.kt:
```diff
@@ -0,0 +1,33 @@
+import ch.nevis.mobile.sdk.api.localdata.Account
+
+/**
+ * [Response] class that indicates an account selection has to be started.
```
I think this description is wrong. It is about account selection instead of transaction confirmation response.

balazs-gerlei replied:
Fixed in 9396c816e5fdd7a55b575e81cc8e29de14c5260b

laszlo-domonkos commented on app/src/main/java/ch/nevis/exampleapp/coroutines/domain/usecase/TransactionConfirmationUseCase.kt:
```diff
@@ -0,0 +1,26 @@
+    /**
+     * Executes the use-case.
+     *
+     * @param account The account that is used for the in-band authentication.
```
It should be out-of-band transaction.

balazs-gerlei replied:
Fixed in 9396c816e5fdd7a55b575e81cc8e29de14c5260b

laszlo-domonkos commented on app/src/main/java/ch/nevis/exampleapp/coroutines/domain/usecase/TransactionConfirmationUseCaseImpl.kt:
```diff
@@ -0,0 +1,29 @@
+
+class TransactionConfirmationUseCaseImpl : TransactionConfirmationUseCase {
+
+    //region InBandAuthenticationUseCase
```
It should be `TransactionConfirmationUseCase`.

balazs-gerlei replied:
Fixed in 9396c816e5fdd7a55b575e81cc8e29de14c5260b

laszlo-domonkos commented on app/src/main/java/ch/nevis/exampleapp/coroutines/ui/transactionConfirmation/parameter/TransactionConfirmationNavigationParameter.kt:
```diff
@@ -16,27 +15,21 @@ import kotlinx.parcelize.Parcelize
     /**
-     * The operation type the account was selection requested for.
-     */
-    val operation: Operation,
```
❓: Just to clarify it. We can get rid of `operation` property because it will be always out-of-band authentication?

balazs-gerlei replied:
Yes, that's what I thought, there was even a comment saying it's always that so I thought what is the reason to pass it around then? Or do you have a reason for it?

laszlo-domonkos replied:
Ok, thanks. Yes, most probably it was unused. I cannot see any reason to keep it at this moment.

balazs-gerlei committed:
NEVISACCESSAPP-6287: Change Fix incorrect code comments

balazs-gerlei committed:
NEVISACCESSAPP-6287: Fix comment for TransactionConfirmationResponse and require the account to exist

- Now that transaction confirmation is after Account Selection, it should always be determined at this point
